src/server.rs

```diff
@@ -26,9 +26,12 @@ impl Server {
         }
     }
 
-    /// This method will run untill the server terminates, `port` defines the listener port.
-    pub fn run(&self, port: u16) -> Result<(), ::error::ServerError> {
-        let addr = format!("0.0.0.0:{}", port).as_str().parse()?;
+    /// This method will run untill the server terminates, `uri` defines the listener uri.
+    pub fn run(&self, uri: &str) -> Result<(), ::error::ServerError> {
+        let url:Uri = uri.parse()?;
+
+        let addr = url.authority_part().expect("The uri passed to launch the server doesn't contain an authority.").as_str().parse()?;
+
```

Comment:
There should be a check for a valid scheme even if only http will works, it'll be more consistent with upcoming version were https would be handled properly. The authority should be validated to be a valid IPv4/v6 too.

Reply:
I added validation for HTTP scheme.  For the authority, it is already validated when we call "parse()". The variable "addr" is already a valid SocketAddr